rpcserver.go

```diff
@@ -476,6 +476,7 @@ func newRPCServer(s *server, macService *macaroons.Service,
 		},
 		FindRoute:       s.chanRouter.FindRoute,
 		ActiveNetParams: activeNetParams.Params,
+		Tower:           s.controlTower,
```

Comment:
can we pass the controltower as a param instead (or make an `rpcserverCfg` to pass in)? Seems unnecessary to increase the server<->rpcserver coupling further, and avoids the server to hold a reference to it.

Reply:
I don't see much harm in this here. Maybe it is good to do some hiding, but I prefer to do it outside of this PR. `r.server` is referenced 113 times in `rpcserver.go`, so it will me a large diff I expect.